### PR TITLE
chore: general simplifications and cleanup

### PR DESCRIPTION
**Pull request by michaelbrewer**
**Issue #, if available:**

## Description of changes:

* Use simplified logic checks
* Use list and dict comprehension
* Remove useless assignments and return value immediately 

**Checklist**

 

* [X] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [X] Update tests
* [X] Update docs
* [X] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by michaelbrewer** [comment]
Nothing functional just code style @heitorlessa

**Review by Nr18 — COMMENTED** [review]
Nice 👌 , @michaelbrewer just a question for my understanding did you replaced the variable assignment by using the `return` for another reason other than a style change?

I have [read](https://code.tutsplus.com/tutorials/understand-how-much-memory-your-python-objects-use--cms-25609) that variable assignment can be memory expensive so just wondering if reducing the memory footprint was a motive of this change as well?

**Comment by michaelbrewer** [comment]
> Nice 👌 , @michaelbrewer just a question for my understanding did you replaced the variable assignment by using the `return` for another reason other than a style change?
> 
> I have [read](https://code.tutsplus.com/tutorials/understand-how-much-memory-your-python-objects-use--cms-25609) that variable assignment can be memory expensive so just wondering if reducing the memory footprint was a motive of this change as well?

It also reduces the number of operations, maybe it is from my old java days when the compiler was not as smart. But to me it seems like an useless assignment that you just return immediately.

File: aws_lambda_powertools/logging/logger.py [hunk 1]
```diff
@@ -291,9 +291,7 @@ def _get_log_level(level: Union[str, int]) -> Union[str, int]:
             return level
 
         log_level: str = level or os.getenv("LOG_LEVEL")
-        log_level = log_level.upper() if log_level is not None else logging.INFO
-
-        return log_level
+        return log_level.upper() if log_level is not None else logging.INFO
```
[the remark this answers is not in the document]
Comment (michaelbrewer) [reply aws_lambda_powertools/logging/logger.py]: Oops @Nr18 will actually break things as `logging.INFO` Is not a str.

[the remark this answers is not in the document]
Comment (Nr18) [reply aws_lambda_powertools/logging/logger.py]:
```
log_level: str = level or os.getenv("LOG_LEVEL", str(logging.INFO))
```
Ah right, that should do it then ;-)

[the remark this answers is not in the document]
Comment (michaelbrewer) [reply aws_lambda_powertools/logging/logger.py]: @Nr18 does not 100% sit right with me. In the case where the log level is not set (which is probably most of the time), then we are taking an `int` casting it to a `str` and then doing an `upper` on it.

[the remark this answers is not in the document]
Comment (michaelbrewer) [reply aws_lambda_powertools/logging/logger.py]: @Nr18 i split up the `if` now, maybe that makes it less confusing?

[the remark this answers is not in the document]
Comment (Nr18) [reply aws_lambda_powertools/logging/logger.py]: Ah right, 👌

**Comment by michaelbrewer** [comment]
@Nr18  does this look good otherwise. Maybe @heitorlessa  can have a check as well

**Review by Nr18 — APPROVED** [review]
LGTM

**Comment by heitorlessa** [comment]
Hey @michaelbrewer thanks a lot! I'll go over that tomorrow - At a quick glance, some are great while others can make debugging difficult (e.g. logger X-Ray trace ID).

File: aws_lambda_powertools/logging/formatter.py [hunk 2]
```diff
@@ -49,9 +49,7 @@ def _build_root_keys(**kwargs):
     @staticmethod
     def _get_latest_trace_id():
         xray_trace_id = os.getenv("_X_AMZN_TRACE_ID")
-        trace_id = xray_trace_id.split(";")[0].replace("Root=", "") if xray_trace_id else None
```
Comment (heitorlessa) [inline aws_lambda_powertools/logging/formatter.py:52]: could you revert this? got caught once trying to debug examples like this, and that's the only reason it has a dedicated assignment despite looking useless at first glance

Comment (heitorlessa) [reply aws_lambda_powertools/logging/formatter.py:52]: nvm, misread the diff - I thought `getenv` was being touched altogether.

**Review by heitorlessa — APPROVED** [review]

**Comment by heitorlessa** [comment]
Misread the tracer getenv line diff so we're good. I left others so we can actually fix the readability in finding a better formatter (Black) configuration -- here's an example where I personally think we lost readability to an extent, as the gain is minimal (perf, mem):

**Before**

```python
        for record in parsed_envelope.Records:
            output.append(self._parse(data=record.Sns.Message, model=model))
        return output
```

**After**

```python
return [self._parse(data=record.Sns.Message, model=model) for record in parsed_envelope.Records]
```

I'll create an issue to ensure these get adjusted automatically next time. 

Thanks **again** @michaelbrewer, much appreciated!

**Pull request merged**